Sources/Plasma/FeatureLib/pfMetalPipeline/plMetalDevice.cpp

```diff
@@ -885,6 +905,12 @@ void plMetalDevice::PopulateTexture(plMetalDevice::TextureRef* tRef, plMipmap* i
             } else {
                 hsAssert(0, "Texture with no image data?\n");
             }
+            
+            if (img->IsCompressed())
+            {
+                // if it's compressed, we created it, we delete it
+                delete img;
+            }
```

Comment:
This should be outside the `for` loop, right?

Reply:
Duh, thanks.